frontend/amundsen_application/static/js/features/ColumnList/ColumnType/parser.ts

```diff
@@ -150,6 +166,55 @@ export function parseNestedType(
   return null;
 }
 
+/**
+ *
+ * @param nestedType
+ */
+export function convertNestedTypeToColumns(
+  nestedType: NestedType,
+  nestedLevel: number = 1
+): TableColumn[] {
+  const { children } = nestedType;
+  const nestedColumns: TableColumn[] = [];
+  children.forEach((child) => {
+    if (typeof child === 'string') {
+      const [columnName, colType] = child.split(' ');
+      if (colType !== undefined) {
+        nestedColumns.push({
+          badges: [],
+          col_type: colType.replace(SEPARATOR_DELIMETER, ''),
+          description: '',
+          name: columnName,
+          sort_order: 0,
+          nested_level: nestedLevel,
+          is_editable: false,
+          stats: [],
+        });
+      }
+    } else {
+      console.log(child);
+      nestedColumns.push({
+        badges: [],
+        col_type: child.col_type || '',
+        description: '',
+        name: child.name || '',
+        sort_order: 0,
+        nested_level: nestedLevel,
+        is_editable: false,
+        stats: [],
+      });
+      const nestedChildren = convertNestedTypeToColumns(child, nestedLevel + 1);
+      nestedColumns.push(...nestedChildren);
+    }
+  });
+  // Need to re-establish the sort order since this is built recursively.
+  nestedColumns.forEach((column, index) => {
+    column.sort_order = index;
+  });
+  return nestedColumns;
```

Comment:
I would make the same comment as above. I hardly make any use of .forEach or .push these days, unless there is a clear performance gain.

Reply:
Normally I would map as well, but this is not a 1:1 mapping. We're reaching into the nested children and flattening them as well.